requirements.txt

```diff
@@ -8,3 +8,4 @@ ROCm/FunctionalPlus@v0.2.18-p0
 ROCm/eigen@3.4.0
 ROCm/frugally-deep@9683d557eb672ee2304f80f6682c51242d748a50
 ROCm/composable_kernel@e4112de7303af1601c6590b964d0df2b6a7f7d32 -DCMAKE_BUILD_TYPE=Release -DINSTANCES_ONLY=ON
+google/googletest@v1.14.0
```

Comment:
For the reviewers (ref: #3005)

- Does MIOpen *require* googletest to function? No, which is why this line was in `dev-requirements.txt` instead.
- Why are we putting it back to `requirements.txt`? (1) cost is relatively minimal (to build googletest); (2) it saves a lot of compatibility check issues.

Reply:
what compatibility checks?

Reply:
This comment has a more detailed response @amberhassaan:
https://github.com/ROCm/MIOpen/issues/3005#issuecomment-2141005498

Realistically, gtest isn't needed in requirements.txt, but anyone building and running the tests for MIOpen would need to install the dev-requirements as well which wasn't a step that was known / happening in all cases. It was working before without installing dev-requirements, since it was finding the gtest installed with conda. Adding it here guarantees it'll work and link correctly with the expected version.

Reply:
I still don't get it. May be @junliume  or @JehandadKhan  can provide some context. Why did we separate `requirements.txt` and `dev-requirements.txt`? if the goal of `requirements.txt` is to enable building `libMIOpen.so` only, then `google-test` is not required and we shouldn't add any such dependency.

Reply:
I think that's exactly right, but the way it was being ran and tested didn't include installing the dev requirements which lead to this issue. I think it's a valid choice to not add gtest, and instead have people change how they install / run the MIOpen test suite, but the issue is that the way they were doing it before just happened to work, and now it won't unless we add gtest to the requirements.txt.